operators/src/adapters/raster_time.rs

```diff
@@ -0,0 +1,379 @@
+        };
+
+        strat.tile_grid_box().number_of_elements()
+    }
+}
+
+impl<T1, T2, St1, St2, F1, F2> Stream for RasterTimeAdapter<T1, T2, St1, St2, F1, F2>
+where
+    T1: Pixel,
+    T2: Pixel,
+    St1: Stream<Item = Result<RasterTile2D<T1>>>,
+    St2: Stream<Item = Result<RasterTile2D<T2>>>,
+    F1: Fn(QueryRectangle) -> St1,
+    F2: Fn(QueryRectangle) -> St2,
+{
+    type Item = Result<(RasterTile2D<T1>, RasterTile2D<T2>)>;
+
+    fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        let RasterTimeAdapterProjection {
+            source_a,
+            source_b,
+            query_rect,
+            time_end,
+            num_spatial_tiles,
+            current_spatial_tile,
+            mut stream,
+            ended,
+        } = self.project();
+
+        if *ended {
+            return Poll::Ready(None);
+        }
+
+        // TODO: handle error
+        let next = ready!(stream.as_mut().poll_next(cx));
+
+        match next {
+            Some((Ok(tile_a), Ok(tile_b))) => {
+                // TODO: calculate at start when tiling info is available before querying first tile
+                if num_spatial_tiles.is_none() {
+                    *num_spatial_tiles = Some(Self::number_of_tiles_in_bbox(
+                        &tile_a.tile_information(),
+                        query_rect.bbox,
+                    ));
+                }
+
+                if *current_spatial_tile >= num_spatial_tiles.expect("checked") {
+                    // time slice ended => query next time slice of sources
+                    let mut next_qrect = *query_rect;
+                    next_qrect.time_interval.start = min(tile_a.time.end, tile_b.time.end);
+                    *time_end = None;
+
+                    stream.set(source_a(next_qrect).zip(source_b(next_qrect)));
+                    *current_spatial_tile = 0;
+                } else {
+                    *current_spatial_tile += 1;
+                }
+
+                Poll::Ready(Some(Ok(Self::align_tiles(tile_a, tile_b))))
+            }
+            Some((Ok(_), Err(e))) | Some((Err(e), Ok(_))) | Some((Err(e), Err(_))) => {
+                *ended = true;
+                Poll::Ready(Some(Err(e)))
+            }
+            None => {
+                *ended = true;
+                Poll::Ready(None)
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::concurrency::ThreadPool;
+    use crate::engine::{
+        ExecutionContext, QueryContext, QueryProcessor, QueryRectangle, RasterOperator,
+        RasterResultDescriptor,
+    };
+    use crate::mock::{MockRasterSource, MockRasterSourceParams};
+    use futures::StreamExt;
+    use geoengine_datatypes::primitives::{BoundingBox2D, SpatialResolution};
+    use geoengine_datatypes::raster::{Grid, RasterDataType};
+    use geoengine_datatypes::spatial_reference::SpatialReference;
+
+    #[tokio::test]
+    async fn adapter() {
```

Comment:
I guess we need a second example where the timestamps don't match. Then, you have to introduce NODATA, right?
I wonder if we have to check the data results themselves in the tests as well and not just the timestamps.

Reply:
The timestamps must match because:
> /// Assumes that raster tiles are contiguous temporally, with no-data-tiles filling gaps.

Reply:
Yes, but you could have two contiguous streams, but the second one ends earlier.

Reply:
you could also have one which starts later.